docs/reference/setup/logging-config.asciidoc

```diff
@@ -194,3 +196,36 @@ files (four rolled logs, and the active log).
 
 You can disable it in the `config/log4j2.properties` file by setting the deprecation
 log level to `error`.
+
+
+[float]
+[[json-logging]]
+=== Json log format
+
+To make parsing elasticsearch logs easier, logs are now printed in a json format.
+This is configured by a log4j layout property `appender.rolling.layout.type = ESJsonLayout`.
+This layout requires a type_name attribute to be set which is used to distinguish
+logs streams when parsing.
+[source,properties]
+--------------------------------------------------
+appender.rolling.layout.type = ESJsonLayout
+appender.rolling.layout.type_name = server
+--------------------------------------------------
+
+Each line contains a single json with set of attributes that are configured in `ESJsonLayout`.
+See this class javadoc for more details.
+When the log event contains a `Throwable`, json will be formatted multiline where first line contains
+all the regular attributes and subsequent lines will contain stacktrace formatted as a json array.
+
+
+NOTE: You can still use your own custom layout. To do that just replace line
+`appender.rolling.layout.type` with the layout of your preference. See sample below:
+[source,properties]
+--------------------------------------------------
+appender.rolling.type = RollingFile
+appender.rolling.name = rolling
+appender.rolling.fileName = ${sys:es.logs.base_path}${sys:file.separator}${sys:es.logs.cluster_name}.log
+appender.rolling.layout.type = PatternLayout
+appender.rolling.layout.pattern = [%d{ISO8601}][%-5p][%-25c{1.}] [%node_name]%marker %.-10000m%n
+appender.rolling.filePattern = ${sys:es.logs.base_path}${sys:file.separator}${sys:es.logs.cluster_name}-%d{yyyy-MM-dd}-%i.log.gz
```

Comment:
Add `_server` suffix for consistency?

Reply:
as discussed not adding to archive names